### PR TITLE
suite/matrix: use math.gcd instead of fractions.gcd

### PR DESCRIPTION
**Pull request by kshtsk**

### CODE REVIEW

**Comment by kshtsk** [comment]
@susebot run deploy

File: requirements.txt [hunk 1]
```diff
@@ -2,71 +2,73 @@
 # This file is autogenerated by pip-compile
 # To update, run:
 #
-#    pip-compile --output-file=requirements.txt
+#    pip-compile --output-file=- requirements.in
```
Comment (tchaikov) [inline requirements.txt:5]: this change bumps lots of packages including `cffi`. would be great if the change is accompanied by the reason why we need to bump up `cffi`.

Comment (kshtsk) [reply requirements.txt:5]: I don't know what exactly want it, python3.9 itself, or the fact it is running Big Sur.

Comment (kshtsk) [reply requirements.txt:5]: if you're asking why this particular line is changed, then it is because update-requirements.sh was used.

Comment (tchaikov) [reply requirements.txt:5]: or you could post the symptom or error message ? and how you connect it to `cffi` to help the reviewers understand this change?

Comment (kshtsk) [reply requirements.txt:5]: @tchaikov the error has a very long trace after this message:
```
Building wheels for collected packages: cffi, gevent
  Building wheel for cffi (setup.py) ... error
  ERROR: Command errored out with exit status 1:
```
.
.
.
```
    clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -DUSE__THREAD -DHAVE_SYNC_SYNCHRONIZE -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/ffi -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/ffi -I/opt/homebrew/include -I/opt/homebrew/opt/openssl@1.1/include -I/opt/homebrew/opt/sqlite/include -I/opt/homebrew/opt/tcl-tk/include -I/Users/kyr/kshtsk/teuthology/virtualenv/include -I/opt/homebrew/Cellar/python@3.9/3.9.2_1/Frameworks/Python.framework/Versions/3.9/include/python3.9 -c c/_cffi_backend.c -o build/temp.macosx-11-arm64-3.9/c/_cffi_backend.o
    c/_cffi_backend.c:6185:5: warning: 'PyEval_InitThreads' is deprecated [-Wdeprecated-declarations]
        PyEval_InitThreads();
        ^
    /opt/homebrew/Cellar/python@3.9/3.9.2_1/Frameworks/Python.framework/Versions/3.9/include/python3.9/ceval.h:130:1: note: 'PyEval_InitThreads' has been explicitly marked deprecated here
    Py_DEPRECATED(3.9) PyAPI_FUNC(void) PyEval_InitThreads(void);
    ^
    /opt/homebrew/Cellar/python@3.9/3.9.2_1/Frameworks/Python.framework/Versions/3.9/include/python3.9/pyport.h:508:54: note: expanded from macro 'Py_DEPRECATED'
    #define Py_DEPRECATED(VERSION_UNUSED) __attribute__((__deprecated__))
                                                         ^
    c/_cffi_backend.c:6245:9: error: implicit declaration of function 'ffi_prep_closure' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
        if (ffi_prep_closure(closure, &cif_descr->cif,
            ^
    1 warning and 1 error generated.
    error: command '/usr/bin/clang' failed with exit code 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /Users/kyr/kshtsk/teuthology/virtualenv/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/tb/jk0cy4c57qj659th07yhns340000gn/T/pip-install-ccbgkb48/cffi_1d01e575fc894776b93086440fce96fb/setup.py'"'"'; __file__='"'"'/private/var/folders/tb/jk0cy4c57qj659th07yhns340000gn/T/pip-install-ccbgkb48/cffi_1d01e575fc894776b93086440fce96fb/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/var/folders/tb/jk0cy4c57qj659th07yhns340000gn/T/pip-record-j3skst66/install-record.txt --single-version-externally-managed --compile --install-headers /Users/kyr/kshtsk/teuthology/virtualenv/include/site/python3.9/cffi Check the logs for full command output.
```

Comment (kshtsk) [reply requirements.txt:5]: updated the message for that commit.

**Review by tchaikov — DISMISSED** [review]

File: requirements.txt (hunk 1 above)
Comment (kshtsk) [reply requirements.txt:5]: @tchaikov I am sorry for being lazy in first place, please give another look if it makes sense.

**Comment by kshtsk** [comment]
> Commit [ffa8ffc](https://github.com/ceph/teuthology/commit/ffa8ffcd43474d8f8aec2546a68c41c4aa6c6a87) is _NOT OK_.
> Check tests results in the Jenkins job: https://ceph-ci.suse.de/job/pr-teuthology-deploy/307/

```
2021-02-26T12:45:17.668 INFO:teuthology.task.ansible.out:ERROR! couldn't resolve module/action 'firewalld'. This often indicates a misspelling, missing collection, or incorrect module path.

The error appears to be in '/home/worker/src/github.com_ceph_ceph-cm-ansible_master/roles/common/tasks/nagios.yml': line 92, column 3, but may
be elsewhere in the file depending on the exact syntax problem.

```
Looks like ansible 3 does not have `firewalld` plugin in the same place it was and moved it to `ansible.posix.firewalld`.

**Comment by kshtsk** [comment]
@susebot run deploy

**Comment by kshtsk** [comment]
@susebot run deploy

**Comment by kshtsk** [comment]
> > Commit [ffa8ffc](https://github.com/ceph/teuthology/commit/ffa8ffcd43474d8f8aec2546a68c41c4aa6c6a87) is _NOT OK_.
> > Check tests results in the Jenkins job: https://ceph-ci.suse.de/job/pr-teuthology-deploy/307/
> 
> ```
> 2021-02-26T12:45:17.668 INFO:teuthology.task.ansible.out:ERROR! couldn't resolve module/action 'firewalld'. This often indicates a misspelling, missing collection, or incorrect module path.
> 
> The error appears to be in '/home/worker/src/github.com_ceph_ceph-cm-ansible_master/roles/common/tasks/nagios.yml': line 92, column 3, but may
> be elsewhere in the file depending on the exact syntax problem.
> ```
> 
> Looks like ansible 3 does not have `firewalld` plugin in the same place it was and moved it to `ansible.posix.firewalld`.

@djgalloway which version of Ansible ceph-cm-ansible currently supports?

**Comment by kshtsk** [comment]
@susebot run deploy

**Review by tchaikov — APPROVED** [review]

**Comment by djgalloway** [comment]
> @djgalloway which version of Ansible ceph-cm-ansible currently supports?

https://github.com/ceph/teuthology/blob/master/requirements.txt#L7

I'm using 2.9.16 locally and also haven't had any issues

**Comment by kshtsk** [comment]
> > @djgalloway which version of Ansible ceph-cm-ansible currently supports?
> 
> https://github.com/ceph/teuthology/blob/master/requirements.txt#L7
> 
> I'm using 2.9.16 locally and also haven't had any issues

@djgalloway Thanks, now I can merge this PR safely.

**Pull request merged**